packages/language-server/src/quick-fix.ts

```diff
@@ -0,0 +1,100 @@
+import { ExecuteCommandParams } from "vscode-languageserver";
+import {
+  Diagnostic,
+  Range as LSPRange,
+  TextDocument,
+  CodeAction,
+  Command,
+  CodeActionKind,
+  TextDocumentEdit,
+  TextEdit,
+} from "vscode-languageserver-types";
+import { getQuickFixIdInfo } from "@ui5-language-assistant/xml-views-quick-fix";
+import { parse, DocumentCstNode } from "@xml-tools/parser";
+import { buildAst, XMLDocument } from "@xml-tools/ast";
+import { LSPRangeToOffsetRange, offsetRangeToLSPRange } from "./range-utils";
+
+export function getQuickFixCodeAction(
+  document: TextDocument,
+  diagnostic: Diagnostic
+): CodeAction | undefined {
+  const documentText = document.getText();
+  // We prefer to parse the document again to avoid cache state handling
+  const { cst, tokenVector } = parse(documentText);
+  const xmlDocAst = buildAst(cst as DocumentCstNode, tokenVector);
+  switch (diagnostic.code) {
+    case 666: {
+      // non stable id
+      return getCodeActionForQuickFixId({
+        document,
+        xmlDocument: xmlDocAst,
+        nonStableIdDiagnostic: diagnostic,
+      });
+    }
+    default:
+      return undefined;
+  }
+}
+
+export function executeCommand(
+  textDocument: TextDocument,
+  params: ExecuteCommandParams
+): TextDocumentEdit[] | undefined {
+  switch (params.command) {
+    case "nonStableIdQuickFix": {
```

Comment:
this (`nonStableIdQuickFix`) looks like a const that should be extracted somewhere, it is used in **three** places.